addons/account/models/chart_template.py

```diff
@@ -293,26 +293,30 @@ def tax_template_changed(tax, template):
                                 if tags:
                                     repartition_line['tag_ids'] = tags
                 elif model_name == 'account.account':
-                    # Point or create xmlid to existing record to avoid duplicate code
                     account = self.ref(xmlid, raise_if_not_found=False)
-                    if not account or (account and account.code != values['code']):
-                        existing_account = self.env['account.account'].search([
-                            ('code', '=', values['code']),
-                            ('company_id', '=', company.id),
-                        ])
-                        if existing_account:
+                    existing_account = self.env['account.account'].search([
+                        ('code', '=', values['code']),
+                        ('company_id', '=', company.id),
+                    ])
+                    if existing_account:
+                        # Point or create xmlid to existing record to avoid duplicate code
+                        if not account or (account and account.code != values['code']):
```

Comment:
```suggestion
                    if existing_account and (not account or account.code != values['code']):
                        # Point or create xmlid to existing record to avoid duplicate code
```
If you keep search() first

Reply:
This would leave the 'reconcile' assignment under the wrong condition. But you're right, I'll see if I can do this while still avoiding unnecesary searches.